### PR TITLE
Fixed agenda numbering. Fixed #1892.

### LINKED ISSUE #1892: Numbering of agenda does not reset item_number of hidden items
How to reproduce?
* create agenda items A, B and C
* click number agenda button -> 1. A, 2. B, 3. C
* set "B" to hidden
* click number agenda button -> 1. A, 2. B, 2. C 

Expected result: 1. A, B, 2. C
The item_number of all hidden items (here "B") has to removed.

### PR DESCRIPTION
**Pull request by normanjaeckel**

### CODE REVIEW

**Comment by ostcar** [comment]
Please fix the tests. The code looks ok

**Comment by normanjaeckel** [comment]
Please review carefully this new code.

File: openslides/agenda/models.py [hunk 1]
```diff
@@ -133,22 +154,31 @@ def number_all(self, numeral_system='arabic'):
...
                 tree_element['item'].save()
                 walk_tree(tree_element['children'], item_number)
 
+        # Start numbering visable agenda items.
         walk_tree(self.get_tree(only_agenda_items=True, include_content=True))
 
+        # Reset number of hidden items.
+        for item in self.get_only_hidden_items():
+            item.item_number = ''
+            item.save()
```
Comment (ostcar) [inline openslides/agenda/models.py]: Proposal:
This is a case where you could work with update()
```python
from openslides.utils.autoupdate import inform_changed_data
hidden_items = self.get_only_hidden_items()  # todo, rename method to get_hidden_items :)
Item.objects.filter(pk__in=[item.pk for item in hidden_items]).update(item_number='')
inform_changed_data(*hidden_items)
```
This is much more efficient if there are a lot of hidden items.

**Comment by ostcar** [comment]
One proposal. Merge if you like

File: openslides/agenda/models.py (hunk 1 above)
Comment (normanjaeckel) [inline openslides/agenda/models.py]: Sorry. Autoupdate does not work with this code.

**Pull request merged**